src/components/tl/ucp/tl_ucp_coll.h

```diff
@@ -415,8 +415,7 @@ ucc_tl_ucp_get_radix_from_range(ucc_tl_ucp_team_t *team,
     radix = ucc_mrange_uint_get(p, msgsize, mem_type);
 
     if (UCC_UUNITS_AUTO == radix) {
-        /* auto selection based on team configuration */
-        return UCC_UUNITS_AUTO_RADIX;
+        return team->opt_radix;
```

Comment:
this assumes that we have single opt_radix for all potential usages of any mrange_uint radix parameter (for any algorithm). I don't think this assumption is true in all cases. originally the fn would return AUTO_RADIX flag that would mean for the caller to use special logic - potentially specific for algorithm for example. that was more clean imho.

Reply:
Yes, but what i'm saying is different. This interface does not allow specifying different optimal radixes for say Bcast and Allreduce.

Reply:
Can you elaborate why? 
From what I understand, if if statement in line 417 is true that UCC_UUNITS_AUTO == radix that means that irrelevant to which coll, radix was not set by user in cmd flags or in ucc.conf, so it can only now have been set by opt radix or not at all in which case team->opt_radix will either be set to something specific or will be UCC_UUNITS_AUTO_RADIX. 
What is the difference if we check which case, and if not set return UCC_UUNITS_AUTO_RADIX directly?

Reply:
Right now you can only compute single value of for opt_radix for tl/ucp team. This opt radix is based currently on your minimizing alg for radix. it is the same for allreduce/bcast etc. So if user keeps AUTO for radix variable, then for all collectives it will use the same team->opt_radix value.

Now assume that you have 2 different algorithms for opt radix calcuation : 1 for allreduce and another for bcast. Then what you could do with the former implementation:
you call ucc_tl_ucp_get_radix_from_range from bcast/allreduce, and get UUNITS_AUTO_RADIX as output. Then you understand (at tl/ucp team level) that you need auto selectin. Then you call custom functions: e.g. tl_ucp_bcast_opt_radix() ro tl_ucp_allreduce_opt_radix() - and they can return different values.

In your PR you can't do that because now ucc_tl_ucp_get_radix_from_range  always returns only 1 possible value team->opt_radix

Reply:
Fixed. Let me know if you have further comments on this.